workflows/crisprseq_screening.nf

```
@@ -134,255 +141,381 @@ workflow CRISPRSEQ_SCREENING {
+            .splitCsv(header: true, sep: ';')
+            .set { ch_drugz }
 
         counts = ch_drugz.combine(ch_counts)
-        DRUGZ (
+        DRUGZ(
             counts
-            )
+        )
         ch_versions = ch_versions.mix(DRUGZ.out.versions)
 
-        if(params.hitselection) {
-            HITSELECTION(DRUGZ.out.per_gene_results,
+        if (params.hitselection) {
+            HITSELECTION(
+                DRUGZ.out.per_gene_results,
                 INITIALISATION_CHANNEL_CREATION_SCREENING.out.biogrid,
                 INITIALISATION_CHANNEL_CREATION_SCREENING.out.hgnc,
-                params.hit_selection_iteration_nb)
+                params.hit_selection_iteration_nb,
+            )
 
             ch_versions = ch_versions.mix(HITSELECTION.out.versions)
         }
-
     }
 
     //
+
+    // Calling of nf-gpt plugin on drugZ, MAGeCK mle or bagel2
+    //
+    if (params.gpt_interpretation && params.gpt_interpretation.split(',').contains('drugz')) {
+        if (params.drugz) {
+            def gpt_drugz_data = DRUGZ.out.per_gene_results.map { meta, genes -> genes }
+            def gpt_drugZ_source = "drugZ"
+            def gpt_drugZ_target_column = 5
+            def gpt_drugZ_mode = "low"
+            GPT_PREPARE_DRUGZ_QUERY(
+                gpt_drugz_data,
+                gpt_drugZ_source,
+                gpt_drugZ_target_column,
+                params.gpt_drugz_gene_amount,
+                gpt_drugZ_mode,
+                params.gpt_drugz_question,
+            )
+
+            GPT_PREPARE_DRUGZ_QUERY.out.query
+                .map { it ->
+                    it.text
+                }
+                .collect()
+                .flatMap { it -> gptPromptForText(it[0]) }
+                .collectFile(name: "${params.outdir}/gpt/gpt_drugz_output.txt", newLine: true, sort: false)
+        }
+        else {
+            error("You specified DrugZ for gpt interpretation, but DrugZ is not running.")
```

I'm not sure we should throw an error here. Wouldn't it be better to show a warning message but continue the executiono f the pipeline?